builder/api/src/main/java/io/helidon/builder/api/Prototype.java

```diff
@@ -0,0 +1,414 @@
+     *     parameter is the builder instance, may have additional parameters, must be void</li>
+     *     <li>{@link Prototype.PrototypeMethod} - to be added to the prototype interface, first parameter
+     *     is the prototype instance, may have additional parameters</li>
+     * </ul>
+     */
+    @Target(ElementType.TYPE)
+    @Retention(RetentionPolicy.CLASS)
+    @interface CustomMethods {
+        /**
+         * Type that implements static methods to be available on the prototype.
+         *
+         * @return type with static methods annotated with {@link Prototype.FactoryMethod}
+         */
+        Class<?> value();
+    }
+
+    /**
+     * Annotated static method of a custom methods type to be added to builder.
+     * First parameter must be the builder type. Return type must be void.
+     */
+    @Target({ElementType.METHOD, ElementType.TYPE})
+    @Retention(RetentionPolicy.CLASS)
+    @interface BuilderMethod {
+    }
+
+    /**
+     * Annotated static method of a custom methods type to be added to prototype interface.
+     * First parameter must be the prototype instance.
+     */
+    @Target({ElementType.METHOD, ElementType.TYPE})
+    @Retention(RetentionPolicy.CLASS)
+    @interface PrototypeMethod {
+    }
+
+    /**
+     * Add additional interfaces to implement by the prototype. Provide correct types (fully qualified) for generics.
+     */
+    @interface Implement {
+        /**
+         * Interfaces to implement, such as {@code java.lang.Comparable<io.helidon.common.types.TypeName>}.
+         *
+         * @return interfaces to implement
+         */
+        String[] value();
+    }
+
+    /**
+     * Mark a getter method as redundant - not important for {@code equals}, {@code hashcode}, and/or {@code toString}.
+     * The generated prototype will ignore the fields for these methods in equals and hashcode methods.
+     * All other fields will be included.
+     * <p>
+     * In case both properties are set to false, it is af this annotation is not present.
+     */
+    @Target(ElementType.METHOD)
+    // note: class retention needed for cases when derived builders are inherited across modules
+    @Retention(RetentionPolicy.CLASS)
+    @interface Redundant {
+        /**
+         * Set to {@code false} to mark this NOT redundant for equals and hashcode.
+         *
+         * @return whether this should be ignored for equals and hashcode
+         */
+        boolean equality() default true;
+
+        /**
+         * Set to {@code false} to mark this NOT redundant for toString.
+         *
+         * @return whether this should be ignored for toString
+         */
+        boolean stringValue() default true;
+    }
+
+    /**
+     * Mark a getter method as confidential - not suitable to be used in clear text in {@code toString} method.
+     * The field will be mentioned (whether it has a value or not), but its value will not be visible.
+     */
+    @Target(ElementType.METHOD)
```

Comment:
Should be on field also.

Reply:
no, it should not. We only support interfaces.